### PR TITLE
Upgrade to latest Keystone and update copy

### PR DESCRIPTION
**Pull request by dcousens**
I don't know if I'm aligned with the idea that we should ship a default example with `access: allowAll`, but I digress.
I have at least removed the impression that access control might be in effect by removing the following code:
```typescript
    ui: {
      // For our starter, we check that someone has session data before letting them see the Admin UI.
      isAccessAllowed: (context) => !!context.session?.data,
    },
```

If we're going to add an `isAccessAllowed` to that effect, then we should add actual access control to everything.
For now I have added the following copy to each of the mandatory `access` definitions:
```typescript
    // WARNING
    //   for this starter project, anyone can create, query, update and delete anything
    //   if you want to prevent random people on the internet from accessing your data,
    //   you can find out more at https://keystonejs.com/docs/guides/auth-and-access-control
```

I might follow that up soon, but at least for now the user has plenty of warning and opportunity to learn what to change and where.

### CODE REVIEW

File: create-keystone-app/starter/package.json [hunk 1]
```diff
@@ -8,22 +8,12 @@
     "build": "keystone build",
     "postinstall": "keystone postinstall"
   },
+  "// @aws-sdk": "temporary dependency until https://github.com/keystonejs/keystone/issues/8023 is resolved",
```
Comment (dcousens) [inline create-keystone-app/starter/package.json:11]: Unfortunate, but we can remove it when https://github.com/keystonejs/keystone/issues/8023 is resolved

File: create-keystone-app/starter/schema.ts [hunk 2]
```diff
@@ -1,94 +1,78 @@
...
+      // by adding isRequired, we enforce that every User should have a name
+      //   if no name is provided, an error will be displayed
       name: text({ validation: { isRequired: true } }),
+
       email: text({
         validation: { isRequired: true },
+        // by adding isIndexed: 'unique', we're saying that no user can have the same
+        // email as another user - this may or may not be a good idea for your project
         isIndexed: 'unique',
-        isFilterable: true,
```
Comment (dcousens) [inline create-keystone-app/starter/schema.ts:53]: The default is `isFilterable: true`

File: create-keystone-app/starter/schema.ts [hunk 3]
```diff
@@ -1,94 +1,78 @@
...
-      // we want a user to have many posts, and we are saying that the user
-      // should be referencable by the 'author' field of posts.
-      // Make sure you read the docs to understand how they work: https://keystonejs.com/docs/guides/relationships#understanding-relationships
+
+      // we can use this field to see what Posts this User has authored
+      //   more on that in the Post list below
       posts: relationship({ ref: 'Post.author', many: true }),
-    },
-    // Here we can configure the Admin UI. We want to show a user's name and posts in the Admin UI
-    ui: {
```
Comment (dcousens) [inline create-keystone-app/starter/schema.ts:64]: I didn't see a reason to include this as is

File: create-keystone-app/starter/schema.ts [hunk 4]
```diff
@@ -1,94 +1,78 @@
...
-      listView: {
-        initialColumns: ['name', 'posts'],
-      },
+
+      createdAt: timestamp({
+        // this sets the timestamp to Date.now() when the user is first created
+        defaultValue: { kind: 'now' }
```
Comment (dcousens) [inline create-keystone-app/starter/schema.ts:58]: As an example of timestamp, this seemed more appropriate compared to `publishDate`

File: create-keystone-app/starter/schema.ts [hunk 5]
```diff
@@ -1,94 +1,78 @@
...
-          { label: 'Published', value: 'published' },
-          { label: 'Draft', value: 'draft' },
-        ],
-        // We want to make sure new posts start off as a draft when they are created
-        defaultValue: 'draft',
-        // fields also have the ability to configure their appearance in the Admin UI
-        ui: {
-          displayMode: 'segmented-control',
-        },
-      }),
```
Comment (dcousens) [inline create-keystone-app/starter/schema.ts:88]: I think we can keep this simpler than a status, if we introduce a `status`, we should do something useful with it.
On that point, we should update the [blog example](https://github.com/keystonejs/keystone/blob/main/examples/blog/schema.ts) to do something with the `status` (like hiding posts that aren't published).

File: create-keystone-app/starter/schema.ts [hunk 6]
```diff
@@ -125,17 +122,27 @@ export const lists: Lists = {
...
+
+  // this last list is our Tag list, it only has a name field for now
   Tag: list({
+    // WARNING
+    //   for this starter project, anyone can create, query, update and delete anything
+    //   if you want to prevent random people on the internet from accessing your data,
+    //   you can find out more at https://keystonejs.com/docs/guides/auth-and-access-control
+    access: allowAll,
+
+    // setting this to isHidden for the user interface prevents this being visible in the Admin UI
```
Comment (dcousens) [inline create-keystone-app/starter/schema.ts:137]:
```suggestion
    // setting this to isHidden for the user interface prevents this list being visible in the Admin UI
```

File: index.test.ts [hunk 7]
```diff
@@ -5,11 +5,12 @@ import path from 'path';
...
 import _treeKill from 'tree-kill';
 import retry from 'async-retry';
+import { randomBytes } from 'crypto';
 
 const treeKill = promisify(_treeKill);
 
-process.env.SESSION_SECRET =
-  'HbGUBzUcVC4ghjg4w4T2Dz4z7dByYCz7GTAUDwaUEEFc2WxkjuPMyqnTtZ4H3hMp';
+// this is passed to Keystone
+process.env.SESSION_SECRET = randomBytes(32).toString('hex')
```
Comment (dcousens) [inline index.test.ts:13]: We should probably only do this for the `production` run-up, if we want to verify that our defaults are working

File: create-keystone-app/starter/auth.ts [hunk 8]
```diff
@@ -1,51 +1,63 @@
...
-
-// Here is a best practice! It's fine to not have provided a session secret in dev,
-// however it should always be there in production.
-if (!sessionSecret) {
-  if (process.env.NODE_ENV === 'production') {
-    throw new Error(
-      'The SESSION_SECRET environment variable must be set in production'
-    );
-  } else {
-    sessionSecret = '-- DEV COOKIE SECRET; CHANGE ME --';
```
Comment (dcousens) [inline create-keystone-app/starter/auth.ts:25]: We shouldn't do this, someone will be burnt :fire: by it, somwhere

Comment (dcousens) [reply create-keystone-app/starter/auth.ts:25]: I guess the problem is this will force someone to login every time they update the schema

**Comment by dcousens** [comment]
I think we need to decide what this example is meant to be.

The balance at the moment is halfway between a template project (like `create-react-app`) and a project to learn Keystone with.
I think Keystone is too complicated to try and box up a starter-project and tutorial in one.

**Review by flexdinesh — APPROVED** [review]

**Pull request merged**